src/transformers/models/whisper/modeling_flax_whisper.py

```diff
@@ -58,6 +59,22 @@
 remat = nn_partitioning.remat
 
 
+def sinusoidal_embedding_init(max_timescale: float = 10000):
```

Comment:
Sorry @gau-nernst, can we not just define one function here? We should move away from defining two functions, where the outer one just calls the inner one directly

Reply:
The reason I do it like this is to allow changing the default `max_timescale` value ("parameterized" function), so that it is on feature-parity with the PyTorch version. If we just define 1 function, there is no way to change `max_timescale` value, since Jax/Keras will only call the function with (shape, dtype) (and additionally RNG for Jax) (technically we can still bypass this by using `functools.partial()`). I follow Jax for this design (https://github.com/google/jax/blob/3247db774ea387098bd9d9049886030dc666cb39/jax/_src/nn/initializers.py#L133-L157). Another way is to make it a class (like Keras does).

Realistically the users won't be able to specify `max_timescale` to the model anyway since we don't expose it. So it would be fine for me to make `max_timescale` as a hard-coded constant also.

Reply:
Yeah if `max_timescales` is not a reachable argument by the user let's just hardcode it. We tend to do this anyway for sinusoidal embeddings: https://github.com/huggingface/transformers/blob/391177441b133645c02181b57370ab12f71b88c4/src/transformers/models/gptj/modeling_flax_gptj.py#L109-L110